### ISSUE TITLE
-Wframe-larger-than in display/dc/dml/dml1_display_rq_dlg_calc.c

### ISSUE THREAD

**Issue by tpgxyz** [ClangBuiltLinux/linux#693]
Building kernel 5.3.1 with LLVM/clang-9.0.0 on i686

```
  CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/dml/dml1_display_rq_dlg_calc.o
BUILDSTDERR: drivers/gpu/drm/amd/amdgpu/../display/dc/dml/dml1_display_rq_dlg_calc.c:979:6: warning: stack frame size of 1212 bytes in function 'dml1_rq_dlg_get_dlg_params' [-Wframe-larger-than=]
BUILDSTDERR: void dml1_rq_dlg_get_dlg_params(
BUILDSTDERR:      ^
BUILDSTDERR: 1 warning generated.
```

**Comment by nathanchance** [comment]
Still reproducible with Linux @ https://git.kernel.org/linus/16fc44d6387e260f4932e9248b985837324705d8 and LLVM @ https://github.com/llvm/llvm-project/commit/1c10201d9660c1d6f43a7226ca7381bfa255105d.

```
$ make -skj"$(nproc)" LLVM=1 LLVM_IAS=1 i386_defconfig

$ scripts/config -e DRM_AMDGPU

$ make -skj"$(nproc)" LLVM=1 LLVM_IAS=1 olddefconfig drivers/gpu/drm/amd/amdgpu/../display/dc/dml/dml1_display_rq_dlg_calc.o
drivers/gpu/drm/amd/amdgpu/../display/dc/dml/dml1_display_rq_dlg_calc.c:979:6: warning: stack frame size of 1132 bytes in function 'dml1_rq_dlg_get_dlg_params' [-Wframe-larger-than=]
void dml1_rq_dlg_get_dlg_params(
     ^
1 warning generated.
```

Running the object file through [`frame-larger-than.py`](https://github.com/ClangBuiltLinux/frame-larger-than):

```
$ python3 $CBL_GIT/frame-larger-than/frame_larger_than.py drivers/gpu/drm/amd/amdgpu/../display/dc/dml/dml1_display_rq_dlg_calc.o dml1_rq_dlg_get_dlg_params
Unsupported type info for DW_TAG_enumeration_type, implement me!
DIE DW_TAG_enumeration_type, size=12, has_children=True
    |DW_AT_type        :  AttributeValue(name='DW_AT_type', form='DW_FORM_ref4', value=75, raw_value=75, offset=894)
    |DW_AT_name        :  AttributeValue(name='DW_AT_name', form='DW_FORM_strp', value=b'cursor_bpp', raw_value=2680, offset=898)
    |DW_AT_byte_size   :  AttributeValue(name='DW_AT_byte_size', form='DW_FORM_data1', value=4, raw_value=4, offset=902)
    |DW_AT_decl_file   :  AttributeValue(name='DW_AT_decl_file', form='DW_FORM_data1', value=1, raw_value=1, offset=903)
    |DW_AT_decl_line   :  AttributeValue(name='DW_AT_decl_line', form='DW_FORM_data1', value=101, raw_value=101, offset=904)

dml1_rq_dlg_get_dlg_params:
	4	unsigned int                  	htotal
	4	unsigned int                  	hblank_end
	8	double                        	pclk_freq_in_mhz
	8	double                        	refclk_freq_in_mhz
	8	double                        	dppclk_freq_in_mhz
	8	double                        	dispclk_freq_in_mhz
	4	unsigned int                  	vblank_start
	4	unsigned int                  	vblank_end
	4	unsigned int                  	min_vblank
	8	double                        	ref_freq_to_pix_freq
	8	double                        	prefetch_xy_calc_in_dcfclk
	8	double                        	t_calc_us
	8	double                        	min_ttu_vblank
	4	unsigned int                  	dlg_vblank_start
	8	double                        	min_dst_y_ttu_vblank
	1	bool                          	mode_422
	4	unsigned int                  	pixel_rate_delay_subtotal
	4	unsigned int                  	vstartup_start
	4	unsigned int                  	dst_y_after_scaler
	4	unsigned int                  	dst_x_after_scaler
	4	unsigned int                  	vready_offset
	4	unsigned int                  	vupdate_width
	4	unsigned int                  	vupdate_offset
	4	unsigned int                  	meta_chunks_per_row_ub_l
	4	unsigned int                  	dpte_groups_per_row_ub_c
	4	unsigned int                  	dpte_bytes_per_row_ub_c
	4	unsigned int                  	swath_width_ub_c
	4	unsigned int                  	swath_height_c
	4	unsigned int                  	meta_bytes_per_row_ub_l
	4	unsigned int                  	meta_pte_bytes_per_frame_ub_l
	4	unsigned int                  	dpte_groups_per_row_ub_l
	4	unsigned int                  	dpte_bytes_per_row_ub_l
	4	unsigned int                  	swath_height_l
	4	unsigned int                  	swath_width_ub_l
	8	double                        	vinit_bot_c
	8	double                        	vinit_bot_l
	8	double                        	vinit_c
	8	double                        	vinit_l
	8	double                        	line_time_in_us
	8	double                        	vratio_c
	8	double                        	vratio_l
	8	double                        	hratios_c
	8	double                        	hratios_l
	4	unsigned int                  	htaps_c
	4	unsigned int                  	htaps_l
	4	unsigned int                  	vp_width_c
	4	unsigned int                  	vp_height_c
	4	unsigned int                  	vp_height_l
	4	unsigned int                  	vp_width_l
	4	unsigned int                  	bytes_per_element_c
	8	double                        	line_wait
	8	double                        	line_o
	8	double                        	line_setup
	8	double                        	line_calc
	8	double                        	dst_y_prefetch
	4	unsigned int                  	meta_row_bytes
	4	unsigned int                  	vm_bytes
	4	unsigned int                  	max_partial_sw_c
	4	unsigned int                  	max_partial_sw_l
	4	unsigned int                  	max_num_sw_c
	4	unsigned int                  	max_num_sw_l
	8	double                        	t_pre_us
	8	double                        	max_vinit_l
	8	double                        	max_vinit_c
	4	unsigned int                  	lsw_l
	4	unsigned int                  	lsw_c
	4	unsigned int                  	sw_bytes_ub_l
	4	unsigned int                  	dpte_row_bytes
	4	unsigned int                  	sw_bytes
	4	unsigned int                  	sw_bytes_ub_c
	8	double                        	prefetch_bw
	8	double                        	flip_bw
	8	double                        	t_vm_us
	8	double                        	t_r0_us
	8	double                        	dst_y_per_row_vblank
	8	double                        	dst_y_per_vm_vblank
	8	double                        	min_dst_y_per_row_vblank
	8	double                        	min_dst_y_per_vm_vblank
	8	double                        	lsw
	8	double                        	vratio_pre_c
	8	double                        	vratio_pre_l
	4	unsigned int                  	req_per_swath_ub_l
	4	unsigned int                  	meta_row_height_l
	4	unsigned int                  	scaler_rec_in_width_c
	4	unsigned int                  	scaler_rec_in_width_l
	4	unsigned int                  	swath_width_pixels_ub_c
	4	unsigned int                  	swath_width_pixels_ub_l
	4	unsigned int                  	req_per_swath_ub_c
	4	unsigned int                  	dpte_row_height_c
	8	double                        	hscale_pixel_rate_l
	8	double                        	hscale_pixel_rate_c
	8	double                        	min_hratio_fact_l
	8	double                        	min_hratio_fact_c
	8	double                        	refcyc_per_line_delivery_pre_l
	8	double                        	refcyc_per_line_delivery_pre_c
	8	double                        	refcyc_per_line_delivery_l
	8	double                        	refcyc_per_line_delivery_c
	8	double                        	refcyc_per_req_delivery_pre_l
	8	double                        	refcyc_per_req_delivery_pre_c
	8	double                        	refcyc_per_req_delivery_l
	8	double                        	refcyc_per_req_delivery_c
	8	double                        	refcyc_per_req_delivery_pre_cur0
	8	double                        	refcyc_per_req_delivery_cur0
	4	unsigned int                  	full_recout_width
	8	double                        	hratios_cur0
	4	unsigned int                  	cur0_src_width
	8	double                        	hactive_cur0
	8	double                        	cur0_req_per_width
	8	double                        	cur0_width_ub
	4	unsigned int                  	cur0_req_width
	4	unsigned int                  	cur0_req_size
	4	None                          	cur0_bpp
	1	bool                          	interlaced
	8	double                        	min_dcfclk_mhz
	1	bool                          	dcc_en
	1	bool                          	dual_plane
	4	unsigned int                  	access_dir
	4	unsigned int                  	bytes_per_element_l
	4	unsigned int                  	dppclk_delay_subtotal
	4	unsigned int                  	dispclk_delay_subtotal
	4	unsigned int                  	dpte_row_height_l
	4	int                           	__ret_warn_on
dml_pow:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
dml_max:
dml_max:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
get_bytes_per_element:
	4	unsigned int                  	ret_val
	4	unsigned int                  	ret_val
dml_max:
dml_max:
	4	int                           	__ret_warn_on
dml_floor:
dml_max:
dml_max:
get_swath_need:
	8	double                        	prefill
	4	unsigned int                  	max_partial_sw_int
	4	int                           	__ret_warn_on
	8	double                        	prefill
	4	unsigned int                  	max_partial_sw_int
dml_floor:
dml_ceil:
is_dual_plane:
	1	bool                          	ret_val
get_swath_need:
	8	double                        	prefill
	4	unsigned int                  	max_partial_sw_int
	4	int                           	__ret_warn_on
	8	double                        	prefill
	4	unsigned int                  	max_partial_sw_int
dml_floor:
dml_ceil:
dml_max:
dml_max:
dml_max:
dml_max:
dml_max:
dml_max:
dml_max:
dml_max:
dml_max:
	4	int                           	__ret_warn_on
dml_pow:
dml_pow:
dml_floor:
dml_pow:
dml_floor:
dml_pow:
	4	int                           	__ret_warn_on
	4	int                           	__ret_warn_on
	4	int                           	__ret_warn_on
get_vratio_pre:
	8	double                        	vratio_pre
	8	double                        	prefill
	8	double                        	tmp0
	8	double                        	vratio_pre
	8	double                        	prefill
dml_floor:
	8	double                        	tmp0
get_vratio_pre:
	8	double                        	vratio_pre
	8	double                        	prefill
	8	double                        	tmp0
	8	double                        	vratio_pre
	8	double                        	prefill
dml_floor:
	8	double                        	tmp0
	4	int                           	__ret_warn_on
dml_pow:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
dml_pow:
dml_pow:
dml_pow:
dml_pow:
dml_pow:
dml_pow:
get_refcyc_per_delivery:
	8	double                        	refcyc_per_delivery
	8	double                        	refcyc_per_delivery
get_refcyc_per_delivery:
	8	double                        	refcyc_per_delivery
	8	double                        	refcyc_per_delivery
dml_floor:
dml_floor:
	4	int                           	__ret_warn_on
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
get_refcyc_per_delivery:
	8	double                        	refcyc_per_delivery
	8	double                        	refcyc_per_delivery
get_refcyc_per_delivery:
	8	double                        	refcyc_per_delivery
	8	double                        	refcyc_per_delivery
dml_floor:
dml_floor:
	4	int                           	__ret_warn_on
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
get_refcyc_per_delivery:
	8	double                        	refcyc_per_delivery
	8	double                        	refcyc_per_delivery
get_refcyc_per_delivery:
	8	double                        	refcyc_per_delivery
	8	double                        	refcyc_per_delivery
dml_pow:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
get_refcyc_per_delivery:
	8	double                        	refcyc_per_delivery
	8	double                        	refcyc_per_delivery
get_refcyc_per_delivery:
	8	double                        	refcyc_per_delivery
	8	double                        	refcyc_per_delivery
dml_pow:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
	4	int                           	__ret_warn_on
	4	unsigned int                  	cur0_bit_per_pixel
dml_ceil:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
	4	int                           	__ret_warn_on
dml_pow:
```

**Comment by nickdesaulniers** [comment]
death by a thousand cuts

**Comment by nathanchance** [comment]
Harry Wentland's recent series fixes this issue: https://lore.kernel.org/amd-gfx/20210909010023.29110-1-harry.wentland@amd.com/

**Comment by nathanchance** [comment]
Harry's series to fix this particular issue was merged in 5.16-rc1: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/log/?h=a62427ef9b55d4b5aec2b08dc9b2d9719572bf57
